### ISSUE TITLE
Weight Decay only in encoder

### ISSUE THREAD

**Issue by pjckoch** [cleinc/bts#95]
Hi @cogaplex-bts,

thanks again for sharing your work.

One question about regularization: is there any specific reason why weight decay is only applied to the encoder, but not to the decoder?

Thanks in advance!

https://github.com/cogaplex-bts/bts/blob/e63ccc00a1d44cc417c745714e3cabe58559f054/pytorch/bts_main.py#L370-L373

**Comment by xuanlongORZ** [comment]
I think this is due to the use of pretrained weights in the encoder while the weights in the decoder are randomly initialed. Thus it is easier for the encoder to be overfitted. There are other strategies like using 0.1*learning_rate for the pretrained encoder and 1*learning_rate for the randomly initialed decoder as used in [Adabins](https://github.com/shariqfarooq123/AdaBins/blob/main/train.py#L149).

**Comment by pjckoch** [comment]
Right, that makes perfect sense. Thanks for explaining @xuanlongORZ !
